moon-store-mac-os/moon-store-mac-os/Views/Login/LoginViewModel.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  LoginViewModel.swift
+//  moon-store-mac-os
+//
+//  Created by Jose Luna on 12/15/24.
+//
+
+import Foundation
+
+final class LoginViewModel: ObservableObject {
+    @Published var email: String = ""
+    @Published var password: String = ""
+    @Published var isLoading: Bool = false
+    @Published var loginSuccess: Bool = false
+    
+    private let authenticationRepository: AuthenticationRepository = .init()
+    
+    func login() {
+        isLoading = true
+        Task { @MainActor in
+            defer {
+                isLoading = false
+                loginSuccess = authenticationRepository.isLoggedUser
+                
+                if loginSuccess {
+                    email = ""
+                    password = ""
+                }
```

Comment:
You can create a private method named `resetCredentialsIfNeeded` to move this logic